File: templated_tests/svelte_calculator/src/lib/utils/calculator.test.js
import { describe, it, expect } from 'vitest';
import { add, subtract, multiply, divide, calculate } from './calculator.js';

describe('Calculator Utils', () => {
	describe('add', () => {
		it('should add two positive numbers', () => {
			expect(add(2, 3)).toBe(5);
		});

		it('should add negative numbers', () => {
			expect(add(-2, -3)).toBe(-5);
		});

		it('should add positive and negative numbers', () => {
			expect(add(5, -3)).toBe(2);
		});
	});

	describe('subtract', () => {
		it('should subtract two positive numbers', () => {
			expect(subtract(5, 3)).toBe(2);
		});

		it('should subtract negative numbers', () => {
			expect(subtract(-2, -3)).toBe(1);
		});
	});

	describe('multiply', () => {
		it('should multiply two positive numbers', () => {
			expect(multiply(3, 4)).toBe(12);
		});

		it('should multiply by zero', () => {
			expect(multiply(5, 0)).toBe(0);
		});

		it('should multiply negative numbers', () => {
			expect(multiply(-2, 3)).toBe(-6);
		});
	});

	describe('divide', () => {
		it('should divide two positive numbers', () => {
			expect(divide(12, 3)).toBe(4);
		});

		it('should handle decimal results', () => {
			expect(divide(5, 2)).toBe(2.5);
		});

		it('should throw error for division by zero', () => {
			expect(() => divide(5, 0)).toThrow('Division by zero');
		});
	});

	describe('calculate', () => {
		it('should perform addition', () => {
			expect(calculate('+', 2, 3)).toBe(5);
		});

		it('should perform subtraction', () => {
			expect(calculate('-', 5, 3)).toBe(2);
		});

		it('should perform multiplication', () => {
			expect(calculate('*', 3, 4)).toBe(12);
		});

		it('should perform division', () => {
			expect(calculate('/', 12, 3)).toBe(4);
		});

		it('should throw error for unknown operation', () => {
			expect(() => calculate('%', 5, 3)).toThrow('Unknown operation: %');
		});
	});
});

File: templated_tests/flutter_calculator/test/calculator_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:flutter_calculator/calculator.dart';

void main() {
  group('Calculator Tests', () {
    late Calculator calculator;

    setUp(() {
      calculator = Calculator();
    });

    test('add should return correct sum', () {
      expect(calculator.add(2, 3), equals(5.0));
      expect(calculator.add(-1, 1), equals(0.0));
      expect(calculator.add(0, 0), equals(0.0));
    });

    test('subtract should return correct difference', () {
      expect(calculator.subtract(5, 3), equals(2.0));
      expect(calculator.subtract(1, 1), equals(0.0));
      expect(calculator.subtract(-2, -3), equals(1.0));
    });

    test('multiply should return correct product', () {
      expect(calculator.multiply(2, 3), equals(6.0));
      expect(calculator.multiply(-2, 3), equals(-6.0));
      expect(calculator.multiply(0, 5), equals(0.0));
    });

    test('divide should return correct quotient', () {
      expect(calculator.divide(6, 2), equals(3.0));
      expect(calculator.divide(-4, 2), equals(-2.0));
      expect(calculator.divide(5, 2), equals(2.5));
    });

    test('divide should throw error for division by zero', () {
      expect(() => calculator.divide(1, 0), throwsArgumentError);
    });

    test('calculate should perform addition', () {
      expect(calculator.calculate('+', 2, 3), equals(5.0));
    });

    test('calculate should perform subtraction', () {
      expect(calculator.calculate('-', 5, 3), equals(2.0));
    });

    test('calculate should perform multiplication', () {
      expect(calculator.calculate('*', 3, 4), equals(12.0));
    });

    test('calculate should perform division', () {
      expect(calculator.calculate('/', 12, 3), equals(4.0));
    });

    test('calculate should throw error for unknown operation', () {
      expect(() => calculator.calculate('%', 5, 3), throwsArgumentError);
    });

    test('calculate should throw error for division by zero', () {
      expect(() => calculator.calculate('/', 5, 0), throwsArgumentError);
    });
  });
}
